geomesa-fs/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/parquet/FilteringIterator.scala

```diff
@@ -53,6 +53,23 @@ class FilteringIterator(partition: String,
   override def getPartition: String = partition
 }
 
+class FilteringTransformIterator(partition: String,
+                                 builder: ParquetReader.Builder[SimpleFeature],
+                                 gtFilter: org.opengis.filter.Filter,
+                                 sft: SimpleFeatureType,
+                                 transform: SimpleFeatureType)
+    extends FilteringIterator(partition, builder, gtFilter) {
+
+  import scala.collection.JavaConversions._
+
+  private val transformIndices = transform.getAttributeDescriptors.map(d => sft.indexOf(d.getLocalName)).toArray
```

Comment:
Are there any order considerations here? 

E.g., if the original SFT had attr1,attr2,attr3, and I wish to project to attr3,attr1, does this handle that?

Reply:
yeah it should handle it - the array would be `(2, 0)` in that case, which would pull them out in order.